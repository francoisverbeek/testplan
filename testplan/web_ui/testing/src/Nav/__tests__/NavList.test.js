import React from 'react';
import {ListGroupItem} from 'reactstrap';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import NavList from '../NavList';

function defaultProps() {
  const entry = {
    uid: '123',
    name: 'test',
    status: 'passed',
    type: 'testplan',
    case_count: {
      passed: 0,
      failed: 0,
    },
  };
  return {
    entries: [entry],
    breadcrumbLength: 1,
    handleNavClick: jest.fn(),
    autoSelect: jest.fn(),
  };
}

describe('NavList', () => {
  let props;
  let mountedNavList;
  const renderNavList = () => {
    if (!mountedNavList) {
      mountedNavList = shallow(
        <NavList {...props} />
      );
    }
    return mountedNavList;
  };

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    mountedNavList = undefined;
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    props.handleNavClick.mockClear();
    props.autoSelect.mockClear();
  });

  it('shallow renders without crashing', () => {
    renderNavList();
  });

  it('shallow renders the correct HTML structure', () => {
    const nav_list = renderNavList();
    expect(nav_list).toMatchSnapshot();
  });

  it('calls autoSelect when mounting and updating', () => {
    const auto_select = props.autoSelect;
    const nav_list = renderNavList();
    expect(auto_select.mock.calls.length).toEqual(1);

    nav_list.setProps({'name': 'test2'});
    nav_list.update();
    expect(auto_select.mock.calls.length).toEqual(2);
  });

  it('calls handleNavClick when nav entry has been clicked', () => {
    const handle_nav_click = props.handleNavClick;
    const nav_list = renderNavList();

    nav_list.find(ListGroupItem).simulate('click');
    expect(handle_nav_click.mock.calls.length).toEqual(1);
  })

});