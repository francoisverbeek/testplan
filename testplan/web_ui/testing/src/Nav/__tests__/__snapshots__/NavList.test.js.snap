// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList shallow renders the correct HTML structure 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavList
    autoSelect={[Function]}
    breadcrumbLength={1}
    entries={
        Array [
            Object {
              "case_count": Object {
                "failed": 0,
                "passed": 0,
              },
              "name": "test",
              "status": "passed",
              "type": "testplan",
              "uid": "123",
            },
          ]
    }
    handleNavClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ListGroup
        tag="ul"
>
        <ListGroupItem
                className="navButton_140hs3a-o_O-navButtonInteract_raa3h5"
                onClick={[Function]}
                tabIndex="1"
                tag="li"
        >
                <NavEntry
                        caseCountFailed={0}
                        caseCountPassed={0}
                        name="test"
                        status="passed"
                        type={undefined}
                />
        </ListGroupItem>
</ListGroup>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListGroupItem
            className="navButton_140hs3a-o_O-navButtonInteract_raa3h5"
            onClick={[Function]}
            tabIndex="1"
            tag="li"
>
            <NavEntry
                        caseCountFailed={0}
                        caseCountPassed={0}
                        name="test"
                        status="passed"
                        type={undefined}
            />
</ListGroupItem>,
        ],
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "123",
          "nodeType": "function",
          "props": Object {
            "children": <NavEntry
              caseCountFailed={0}
              caseCountPassed={0}
              name="test"
              status="passed"
              type={undefined}
/>,
            "className": "navButton_140hs3a-o_O-navButtonInteract_raa3h5",
            "onClick": [Function],
            "tabIndex": "1",
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "caseCountFailed": 0,
              "caseCountPassed": 0,
              "name": "test",
              "status": "passed",
              "type": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ListGroup
          tag="ul"
>
          <ListGroupItem
                    className="navButton_140hs3a-o_O-navButtonInteract_raa3h5"
                    onClick={[Function]}
                    tabIndex="1"
                    tag="li"
          >
                    <NavEntry
                              caseCountFailed={0}
                              caseCountPassed={0}
                              name="test"
                              status="passed"
                              type={undefined}
                    />
          </ListGroupItem>
</ListGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ListGroupItem
              className="navButton_140hs3a-o_O-navButtonInteract_raa3h5"
              onClick={[Function]}
              tabIndex="1"
              tag="li"
>
              <NavEntry
                            caseCountFailed={0}
                            caseCountPassed={0}
                            name="test"
                            status="passed"
                            type={undefined}
              />
</ListGroupItem>,
          ],
          "tag": "ul",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "123",
            "nodeType": "function",
            "props": Object {
              "children": <NavEntry
                caseCountFailed={0}
                caseCountPassed={0}
                name="test"
                status="passed"
                type={undefined}
/>,
              "className": "navButton_140hs3a-o_O-navButtonInteract_raa3h5",
              "onClick": [Function],
              "tabIndex": "1",
              "tag": "li",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "caseCountFailed": 0,
                "caseCountPassed": 0,
                "name": "test",
                "status": "passed",
                "type": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
